_posts/2021-03-31-relaxing-requirements-of-moved-from-objects.md

```diff
@@ -167,7 +185,7 @@ Given the above general requirement, we can then specify what operations must ho
 
 ### Option 1
 
-Option 1 requires that a moved-from object can be used as an rhs argument to move-assignment only in the case that the object has been moved from and it is a self-move-assignment. It introduces a _moved-from-value_ to discuss the properties of the moved-from object without specifying a specific value and requires that self-move-assignment for the moved-from object is valid. The wording allows for `swap(a, a)` without allowing `a = move(a)` in general.
+Option 1 requires that a moved-from object can be used as an rhs argument to move-assignment only in the case of self-move-assignment. It introduces a _moved-from-value_ to discuss the properties of the moved-from object without specifying a specific value and requires that self-move-assignment for the moved-from object is valid. The wording allows for `swap(a, a)` without allowing `move(a), b = move(a)` in general.
```

Comment:
I did mean `a = move(a)` although your phrasing is also true but this was to call out that only self-move-assignment for a moved-from object is required and not self-move-assignment in general.

Reply:
Oh, wow… I see now.  The original wording is really easy to misread as self-contradictory. Allow me to suggest:

```suggestion
Option 1 guarantees that the library will only perform self-move-assignment on values that have already been moved from. It introduces a _moved-from-value_ to discuss the properties of the moved-from object without specifying a specific value and requires that self-move-assignment for the moved-from object is valid. The wording allows for `swap(a, a)` without allowing `a = move(a)` in general.
```

I'm not sure if I got the phrasing right; is this about restricting the library?  Otherwise, maybe something like this:

```suggestion
Option 1 forbids self-move-assignment except on values that have already been moved from. It introduces a _moved-from-value_ to discuss the properties of the moved-from object without specifying a specific value and requires that self-move-assignment for the moved-from object is valid. The wording allows for `swap(a, a)` without allowing `a = move(a)` in general.
```